lib/cc/engine/analyzers/engine_config.rb

```diff
@@ -55,6 +55,10 @@ def fetch_language(language)
           end
         end
 
+        def patterns_for(language, fallbacks)
```

Comment:
The integration-y test you have is great. Would it be reasonable to get some unit tests as well? For example, in addition to the happy path, it'd be nice to have a test that a malformed config like:

```
ruby:
  patterns: "*.rb"
```

does the right thing. (I think your code does handle this quite nicely, but guarding against future regressions with a test would be nice.)